core-plugins/src/main/java/co/cask/hydrator/plugin/common/FileSetUtil.java

```diff
@@ -83,18 +86,57 @@ public static void configureParquetFileSet(String configuredSchema, FileSetPrope
   }
 
   /**
+   * Configure a file set to use ORC file format with a given schema. The schema is parsed
+   * validated and converted into a Hive schema which is compatible with ORC format. The file set is configured to use
+   * ORC input and output format, and also configured for Explore to use Hive. The schema is added
+   * to the file set properties in all the different required ways:
+   * <ul>
+   * <li>As a top-level dataset property;</li>
+   * <li>As the schema for the input and output format;</li>
+   * <li>As the schema to be used by the ORC serde (which is used by Hive).</li>
+   * </ul>
+   * @param configuredSchema the original schema configured for the table
+   * @param properties       a builder for the file set properties
+   */
+  public static void configureORCFileSet(String configuredSchema, FileSetProperties.Builder properties)
+    throws IOException, UnsupportedTypeException {
```

Comment:
instead of throwing these exceptions, catch them in this method and throw an IllegalArgumentException with an informative message about what is invalid and why.